File: OIEx/K_Cesar.cc
#include <iostream>
#include <string>

using namespace std;

string letras_mayuscula = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
string letras = "abcdefghijklmnopqrstuvwxyz";
int casos_num;
int clave;
string problema;

int main()
{
    cin >> casos_num;

    for (int i = 0; i < casos_num; i++)
    {
        cin >> clave;
        cin >> problema;

        for (int e = 0; e < problema.length(); e++)
        {
            if (isupper(problema[e]))
            {
                problema[e] = 0;
            }
        }
    }

    return 0;
}

File: OIEx/K_Cesar.py
letras = "abcdefghijklmnopqrstuvwxyz"

casos = int(input())

for i in range(casos):
    output = ""
    clave = int(input())
    problema = input()

    for e in range(len(problema)):
        
        letra = problema[e]

        if letra not in letras and letra not in letras.upper(): output += letra; continue
        
        if letra.islower():
            if letras.find(letra.lower())+clave < len(letras)-1:
                output += letras[letras.find(letra.lower())-clave]
            else:
                output += letras[letras.find(letra.lower())-clave]
                

        elif letra.isupper():
            if letras.find(letra.lower())+clave < len(letras)-1:
                output += letras[letras.find(letra.lower())-clave].upper()
            else:
                output += letras[letras.find(letra.lower())-clave].upper()

        
    print(output)
